scio-smb/src/main/scala/com/spotify/scio/smb/syntax/SortMergeBucketScioContextSyntax.scala

```diff
@@ -141,9 +143,10 @@ final class SortedBucketScioContext(@transient private val self: ScioContext) ex
   def sortMergeCoGroup[K: Coder, A: Coder, B: Coder](
     keyClass: Class[K],
     a: SortedBucketIO.Read[A],
-    b: SortedBucketIO.Read[B]
+    b: SortedBucketIO.Read[B],
+    targetParallelism: TargetParallelism
```

Comment:
not super happy with this API change (i.e not providing a default here) but scalac won't allow multiple overloaded methods with default values :/ any suggestions?

Reply:
Does overloading method without defaults work, or does it cause erasure issues?

```
def sMCG(k, a, b) ...
def sMCG(k, a, b, tp) ...
def sMCG(k, a, b, c) ...
def sMCG(k, a, b, c, tp) ...
```

Otherwise, what about making `TargetParallelism` an implicit argument in all methods, and have a default instance of `min` (old default behavior?). Similar to how executors are used in `scala.concurrent`. But that is if we don't add other params later.

Reply:
oh cool that worked! 🙌